### ISSUE TITLE
Where with :and should remove `nil` clauses

### ISSUE THREAD

**Issue by arichiardi** [jkk/honeysql#203]
Simple example and maybe I am doing something wrong:

```clojure
cljs.user=> (-> (sqlh/where [:= :domain "ars"] [:= :entity "assar"]
                            (when false [:= :key "asr"])) 
                (sql/format  :parameterizer :postgresql))
["WHERE (domain = $1 AND entity = $2 AND NULL)" "ars" "assar"]
```

Probably the last `null` should not be generated.

**Comment by MichaelBlume** [comment]
I think I agree that it'd be better if the NULL wasn't generated

**Comment by arichiardi** [comment]
Solved by #205.
